### ISSUE TITLE
proposal: refactor `refund` from `RwTable` into StepStateTransition

### ISSUE THREAD

**Issue by lispc** [appliedzkp/zkevm-circuits#395]
`refund` is very similar to `gas_left`. For every step, it is only need to check diff rather than value. Maybe StepStateTransition is a more suitable method for handling refund than `RwTable`?

**Comment by CPerezz** [comment]
ping @han0110 

Doesn't seem a crazy idea. But maybe I'm missing something.

**Comment by lispc** [comment]
I thought so. But later I realized that refund may be revert. Currently I am not sure whether it is a good idea. ( I am not very familiar with the overall reversion design)

**Comment by han0110** [comment]
It makes sense to me to track it in `StepState` directly. Although it's reversible thing in EVM, but it doesn't affect future execution, so it's kind of `write_only_persistent` data (like `Log` and `Destructed`), then we only need to add value to this field when `is_persistent == True`.

Another thought in my mind was to keep it in `rw_table`, and also move `memory_word_size` and `state_write_counter` into `rw_table` because they are rarely changed. Until now I still don't see which one is more efficient (or has more space to be optimized), but it seems easier to reason by just tracking it.

**Comment by CPerezz** [comment]
ping @lispc 

Do you want to give it a try?

**Comment by lispc** [comment]
yes you can assign to me. (But i may only do this two weeks later, although i think it is not urgent)
